tests/integration/devfile/cmd_devfile_describe_test.go

```diff
@@ -82,4 +83,32 @@ var _ = Describe("odo devfile describe command tests", func() {
 			helper.CmdShouldPass("odo", "delete", "-f", "--all", "--context", commonVar.Context)
 		})
 	})
+
+	Context("when running odo describe", func() {
+		It("should show json output for working cluster", func() {
+			helper.CmdShouldPass("odo", "create", "nodejs", "--context", commonVar.Context)
+			output := helper.CmdShouldPass("odo", "describe", "--context", commonVar.Context, "-o", "json")
+			values := gjson.GetMany(output, "kind", "metadata.name", "status.state")
+			Expect(helper.GjsonMatcher(values, []string{"Component", "nodejs", "Not Pushed"})).To(Equal(true))
+		})
+
+		It("should show json output for working cluster for a pushed component", func() {
```

Comment:
Why don't we just do `odo push` in [Spec](https://github.com/openshift/odo/pull/4501/files#diff-7373622ac627f4f7b03b7aac291951160b62c57c045121d79c3839e625c19285R88) and validates the behaviour

Reply:
done